apps/ios/SimpleX NSE/NotificationService.swift

```diff
@@ -248,100 +287,144 @@ class NotificationService: UNNotificationServiceExtension {
-                    let started = NSEThreads.queue.sync {
-                        let canStart = checkCanStart(id)
-                        if let t = threadId { logger.debug("NotificationService thread \(t, privacy: .private): receiveNtfMessages: can start: \(canStart)") }
-                        if canStart {
-                            processDroppedNotifications(id)
-                            expectedMessages[id]?.startedProcessingNewMsgs = true
-                            expectedMessages[id]?.shouldProcessNtf = true
+                // uncomment localDisplayName in ConnectionEntity
+                // logger.debug("NotificationService: receiveNtfMessages: apiGetNtfConns ntfConns \(String(describing: ntfConns.map { $0.connEntity.localDisplayName }))")
+
+                // keep track of expected messages - they will be delivered to the reception loop in this chain:
+                // 0. apiGetConnNtfMessages or getConnNtfMessage get messages from the server for passed connection entities.
+                // We don't know in advance which chat events will be delivered from app core for a given notification,
+                // it may be a message, but it can also be contact request, various protocol confirmations, calls, etc.,
+                // this function only returns metadata for the expected chat events.
+                // This metadata is correlated with .msgInfo core event - this marker allows determining when some message completed processing.
+                // 1. receiveMessages: singleton loop receiving events from core.
+                // 2. receivedMsgNtf: maps core events to notification events.
+                // 3. NSEThreads.shared.processNotification: chooses which notification service instance in the current process should process notification.
+                // While most of the time we observe that notifications are delivered sequentially, nothing in the documentation confirms it is sequential,
+                // and from various sources it follows that each instance executes in its own thread, so concurrency is expected.
+                // 4. processReceivedNtf: one of the instances of NSE processes notification event, deciding whether to request further messages
+                // for a given connection entity (via getConnNtfMessage) or that the correct message was received and notification can be deliverd (deliverBestAttemptNtf).
+                // It is based on .msgInfo markers that indicate that message with a given timestamp was processed.
+                // 5. deliverBestAttemptNtf: is called multiple times, once each connection receives enough messages (based on .msgInfo marker).
+                // If further messages are expected, this function does nothing (unless it is called with urgent flag from timeout/expiration handlers).
+                let expectedMsgs: [ExpectedMessage] = ntfConns.compactMap(mkExpectedMessage)
```

Comment:
maybe move above comment outside of function? it's starting to get hard to follow logic in code at this point